jsonnet/kube-thanos/kube-thanos-store.libsonnet

```diff
@@ -78,7 +79,11 @@ function(params) {
             { config+: { service_name: defaults.name } } + ts.config.tracing
           ),
         ] else []
-      ),
+      ) + util.withOptionalArgs(ts.config, {
+        'max-time': 'maxTime',
+        'min-time': 'minTime',
+        'store.grpc.series-sample-limit': 'storeGRPCSeriesSampleLimit',
+      }),
```

Comment:
I think it should be possible to add these flags only of they are not null.
Something along the lines of this:
```jsonnet
     args: [
       ...
      ] + (
        if ts.config.maxTime != null then ['--max-time=' + ts.config.maxTime] else []
      ) + (
```

What this does is either merge `[]` to the `args` array or merge an array with the flag `['--max-time=foo']`. The first case doesn't change the args array and is therefore a no-op.

Reply:
I could add the 3 flags with 3 if-statements (and I see there are similar constructs in this file already) but that seems quite verbose to me 😅 

If we end up adding support for all thanos flags, each component will have a sea of these if-statements. I think we need something more general - and now that I look at the open PR list, it looks like the WIP in https://github.com/thanos-io/kube-thanos/pull/184 might be working towards that. 🤔

Reply:
@metalmatze sorry for the impatience, but I don't want this to get stale if possible. what do you think of my comment above?